### ISSUE TITLE
Sorry, wrong add-on

### ISSUE THREAD

**Issue by alexolog** [EionRobb/skype4pidgin#497]
Running on Windows.

Yesterday everything worked OK
Today Pidgin does not connect to skype, claiming "Failed getting Skype token"
Both the desktop client and the web interface work OK.
I deleted all the cached certificates, but it didn't help.

Please advise.

**Comment by alexolog** [comment]
Tried to run the script from #479 and got this:

b'{"skypetoken":"<long string removed>","expiresIn":86400}'

**Comment by EionRobb** [comment]
Can you try logging into web.skype.com in an incognito browser window to see if there's any extra steps that you need to approve?

**Comment by alexolog** [comment]
Using Firefox's private window works fine on web.skype.com

**Comment by dequis** [comment]
> `b'{"skypetoken":"<long string removed>","expiresIn":86400}'`

Is `<long string removed>` something you wrote there? Did you modify the script in any way? Seems like something went wrong somehow

**Comment by alexolog** [comment]
Nope, used the script as is and only replaced the string I copied from the output.
In particular, the string DID NOT say 'yep looks good'

I tried with a couple of other skype account and got the same result (different string though)

**Comment by dequis** [comment]
Ehhh probably harmless, looks like some issue in the interaction between python3's urllib and the headers sent by the server which makes it return bytes instead of unicode. I'm pretty sure it worked before (the script has explicit python2/python3 support), but I could be wrong. This doesn't mean the method is broken, only a slight bug in the testing script.

Either way irrelevant to this ticket.

**Comment by alexolog** [comment]
Whatever the problem is, it happened quite suddenly, and wouldn't go away.

**Comment by dequis** [comment]
Debug log?

**Comment by alexolog** [comment]
https://paste.ee/p/Vacjo

Account name changed to protect the guilty.

**Comment by alexolog** [comment]
I tried connecting from another computer and it worked.

Comparing the configuration files showed that the failing machine did not have a 'refresh-token' setting in its accounts.xml file while the other machine did.

<account>
	<protocol>prpl-skypeweb</protocol>
	<settings>
		<setting name='refresh-token' type='string'>long string here</setting>
	</settings>
</account>

When I removed the refresh-token from the other machine, it started failing as well.  When I copied it to the first machine, it started working.

**Comment by dreamlayers** [comment]
When first connecting from a Pidgin install I haven't used a while, I often get "Failed getting Skype token" at the bottom of the Pidgin window. After I click on the button to re-enable the account, it successfully connects, though ti may take a bit longer than it should.

**Comment by intika** [comment]
This could may be help... have a look 
https://github.com/EionRobb/skype4pidgin/issues/474#issuecomment-263105854

**Comment by veritas1980** [comment]
I am getting this error starting this morning.

**Comment by xvitaly** [comment]
> I am getting this error starting this morning.

Yes, same problem since today. In debug log I can see HTML with this error:
```
Sorry, something went wrong. Please try again.
```

**Comment by veritas1980** [comment]
http://heartbeat.skype.com/2016/12/skype_multiple_service_outage.html there's an outage.

**Comment by ViperGeek** [comment]
In testing authentication via an Incognito window to https://web.skype.com I confirmed -- now that Skype has been fully Microsoft-ized -- two-factor authentication will cause the "Failed getting Skype token" error:

https://account.live.com/proofs/Manage ("Two-step verification")

So now you must choose: a secure 2FA Microsoft account or using Skype via Pidgin.

- Dave

**Comment by dequis** [comment]
@ViperGeek #385

**Comment by cexikitin** [comment]
@EionRobb is this still being looked at? I'll often get the "Failed getting skype token" but hitting reenable fixes it 100% of the time. Without any notification that the connection is down I can be disconnected for hours.

**Comment by dequis** [comment]
The last commit adds a checkbox in the advanced tab, "Use alternative login method", which might help.

**Comment by alexolog** [comment]
What is the "alternative" method and when it is recommended to use it?

**Comment by dequis** [comment]
See #556 for implementation details. I'd like to make it the default for everyone soon, but it's opt-in for now just in case it doesn't work for some people.

**Comment by alexolog** [comment]
Thanks!  Is there a Windows build?

**Comment by EionRobb** [comment]
@alexolog I've just updated the dll at https://eion.robbmob.com/libskypeweb.dll but it's not showing the account option yet

**Comment by alexolog** [comment]
Thanks Eion!

I know that you are hosting Win32 DLLS for several plugins on your site.  Would it be possible to have automated nightly builds for them?  Maybe something like http://beam.to/fchat-pidgin ?  And if so, would it be possible to have them on a page linked from the main Pidgin site?

**Comment by dequis** [comment]
>but it's not showing the account option yet

Wait, why.

>Last-Modified: Sun, 18 Jun 2017 08:13:18 GMT

You sure about that?

**Comment by EionRobb** [comment]
>but it's not showing the account option yet

... because I was silly and didn't update my code locally. @alexolog if you want to grab the latest dll and give it another try?

**Comment by ViperGeek** [comment]
I just updated to the latest https://eion.robbmob.com/libskypeweb.dll (no version in the DLL header; SHA-1: 6957AB8A67791E907C98580292954ADEBEE1F7E2) and am also getting:

Failed getting Magic T value

- Dave

**Comment by alexolog** [comment]
The new version has 4 checkboxes in the advanced tab, and the top one has no text associated with it.

![skype](https://user-images.githubusercontent.com/2112513/27556896-879d79e8-5a85-11e7-85d8-0fef3820b0d6.JPG)

**Comment by dequis** [comment]
>Login error: Invalid value "ViperGeek" for username param "username" (code 40002)
>
>UPDATE: After converting my username to lowercase, I'm in!

Ooooh, nice find, thanks!

>the top one has no text associated with it

Yeah, that's intentional. It made sense at one point. It doesn't do anything now.

**Comment by alexolog** [comment]
So can it be removed?
